src/open_weather_map/client.rs

```diff
@@ -43,7 +53,37 @@ impl WeatherApiClient {
         Self::decode_response(response)
     }
 
-    pub fn decode_response(mut response: reqwest::Response) -> Result<Weather, ClientError> {
+    pub async fn fetch_weekly(&self, lat: f64, lon: f64) -> Result<WeatherForecast, ClientError> {
+        let request_url = format!(
+            "https://api.openweathermap.org/data/2.5/forecast?lat={}&lon={}&appid={}&units={}&lang={}&cnt={}",
+            lat,
+            lon,
+            OPEN_WEATHER_MAP_API_TOKEN.as_str(),
+            UNITS,
+            LANG,
+            9,
+        );
+
+        let response = self.client.get(&request_url).send()?;
+
+        Self::decode_weekly_response(response)
+    }
+
+    fn decode_weekly_response(
+        mut response: reqwest::Response,
+    ) -> Result<WeatherForecast, ClientError> {
+        let status_code = response.status().as_u16();
+        let string_response = response.text()?;
+
+        if status_code == 200 {
+            let json_result: WeatherForecast = serde_json::from_str(&string_response).unwrap();
```

Comment:
let's not use unwrap?

Reply:
[update to fang 10 and delete an unwrap](https://github.com/pxp9/weather_bot_rust/pull/29/commits/cd8fd775715b195ec1949e1ad8c69355a8c9b8d8)